enclave/crypto/openssl/init.c

```diff
@@ -25,6 +37,54 @@ static int _initialize_symcrypt_engine()
     return (result == OE_SYMCRYPT_ENGINE_SUCCESS) ? 1 : 0;
 }
 
+#if OECRYPTO_OPENSSL_VER >= 3
+static OSSL_PROVIDER* _default_prov;
+static OSSL_PROVIDER* _scossl_prov;
+
+static void _finalize_prov(void)
+{
+    if (_default_prov)
+    {
+        OSSL_PROVIDER_unload(_default_prov);
+        _default_prov = NULL;
+    }
+    if (_scossl_prov)
+    {
+        OSSL_PROVIDER_unload(_scossl_prov);
+        _scossl_prov = NULL;
+    }
+}
+
+static int _initialize_symcrypt_provider()
+{
+    int result = 1;
+
+    // The default provider is loaded first to make sure that OpenSSL can
+    // fall back on it when operations not supported by SymCrypt are called
+    _default_prov = OSSL_PROVIDER_load(NULL, "default");
+
+    if (_default_prov == NULL)
+        result = 0;
+
+    result &= OSSL_PROVIDER_add_builtin(
+        NULL, "symcrypt_provider", OSSL_provider_init);
+    _scossl_prov = OSSL_PROVIDER_load(NULL, "symcrypt_provider");
+
+    if (_scossl_prov == NULL)
+        result &= 0;
```

Comment:
What's use of `&=` here?

Reply:
It's to ensure that all the condition checks in the function must pass in order for result to be 1